pkg/operator/ceph/nfs/config.go

```diff
@@ -126,3 +133,231 @@ RADOS_URLS {
 %url	` + url + `
 `
 }
+
+func ganeshaKrbConfigBlock(kerberosSpec *cephv1.KerberosSpec) string {
+	return fmt.Sprintf(`NFS_KRB5 {
+	PrincipalName = "%s" ;
+	KeytabPath = /etc/krb5.keytab ;
+	Active_krb5 = YES ;
+}
+`, kerberosSpec.GetPrincipalName())
+}
+
+func ganeshaConfigIncludeKrbBlock(nfs *cephv1.CephNFS, radosObjectName string) string {
+	// don't use sprintf b/c %u on front makes compiler confused
+	return `%url "rados://` + nfs.Spec.RADOS.Pool + `/` + nfs.Spec.RADOS.Namespace + `/` + radosObjectName + `"` + "\n\n"
+}
+
+func (r *ReconcileCephNFS) setRadosConfig(nfs *cephv1.CephNFS) error {
+	if nfs.Spec.Security.KerberosEnabled() {
+		return setKerberosRadosConfig(r.context, r.clusterInfo, nfs)
+	}
+	return removeKerberosRadosConfig(r.context, r.clusterInfo, nfs)
+}
+
+func setKerberosRadosConfig(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo, nfs *cephv1.CephNFS) error {
+	radosPool := nfs.Spec.RADOS.Pool
+	radosNs := nfs.Spec.RADOS.Namespace
+	radosInfoStr := fmt.Sprintf("rados://%q/%q/", radosPool, radosNs)
```

Comment:
How about some logging in some of these methods so we know when locks are being added/removed, or the config is being updated, etc?